source/material_model/rheology/friction_options.cc

```diff
@@ -0,0 +1,849 @@
+        else if (prm.get ("Friction dependence mechanism") == "rate and state dependent friction plus linear slip weakening")
+          friction_dependence_mechanism = rate_and_state_dependent_friction_plus_linear_slip_weakening;
+        else if (prm.get ("Friction dependence mechanism") == "slip rate dependent rate and state dependent friction")
+          friction_dependence_mechanism = slip_rate_dependent_rate_and_state_dependent_friction;
+        else if (prm.get ("Friction dependence mechanism") == "regularized rate and state friction")
+          friction_dependence_mechanism = regularized_rate_and_state_friction;
+        else
+          AssertThrow(false, ExcMessage("Not a valid friction dependence option!"));
+
+        if ((friction_dependence_mechanism == slip_rate_dependent_rate_and_state_dependent_friction)
+            |(friction_dependence_mechanism == regularized_rate_and_state_friction)
+            |(friction_dependence_mechanism == rate_and_state_dependent_friction_plus_linear_slip_weakening)
+            |( friction_dependence_mechanism == rate_and_state_dependent_friction))
+          {
+            // Currently, it only makes sense to use a state variable when the nonlinear solver
+            // scheme does a single Advection iteration and at minimum one Stokes iteration, as
+            // the state variable is implemented as a material field. More
+            // than one nonlinear Advection iteration will produce an unrealistic values
+            // for the state variable.
+            AssertThrow((this->get_parameters().nonlinear_solver ==
+                         Parameters<dim>::NonlinearSolver::single_Advection_single_Stokes
+                         ||
+                         this->get_parameters().nonlinear_solver ==
+                         Parameters<dim>::NonlinearSolver::single_Advection_iterated_Stokes
+                         ||
+                         this->get_parameters().nonlinear_solver ==
+                         Parameters<dim>::NonlinearSolver::single_Advection_iterated_Newton_Stokes
+                         ||
+                         this->get_parameters().nonlinear_solver ==
+                         Parameters<dim>::NonlinearSolver::single_Advection_iterated_defect_correction_Stokes),
+                        ExcMessage("The rate and state friction will only work with the nonlinear "
+                                   "solver schemes 'single Advection, single Stokes' and "
+                                   "'single Advection, iterated Stokes', 'single Advection, "
+                                   "iterated defect correction Stokes', 'ginle advection, iterated Newton Stokes'"));
+          }
+
+        // Dynamic friction parameters
+        dynamic_characteristic_strain_rate = prm.get_double("Dynamic characteristic strain rate");
+
+        if (prm.get ("Dynamic angles of internal friction") == "9999")
```

Comment:
I used this check to set the default value to the internal angle of friction so that it would not matter if not set. But I agree, that a check for the mechanism would be nice

Reply:
I could also leave it out completely. I think it was from before I introduced the switch for friction_mechanism? so it was easier to use by setting the default of the dynamic angle to the internal angle. I will take it out then, you are right, that it is not needed anymore.